experimental/dds/tree2/src/shared-tree-core/branch.ts

```diff
@@ -54,35 +54,33 @@ export type SharedTreeBranchChange<TChange> =
 	  };
 
 /**
- * Returns true iff the given {@link SharedTreeBranchChange} is the result of a rebase operation.
+ * Returns the operation that caused the the given {@link SharedTreeBranchChange}.
  */
-export function isRebaseChange(
+export function getChangeReplaceType(
 	change: SharedTreeBranchChange<unknown> & { type: "replace" },
-): boolean {
-	return !isTransactionCommitChange(change);
-}
+): "transactionCommit" | "rebase" {
+	// The "replace" variant of the change event is emitted by two operations: committing a transaction and doing a rebase.
+	// Committing a transaction will always remove one or more commits (the commits that were squashed),
+	// and will add exactly one new commit (the squash commit).
+	if (change.removedCommits.length === 0 || change.newCommits.length !== 1) {
+		return "rebase";
+	}
 
-/**
- * Returns true iff the given {@link SharedTreeBranchChange} is the result of a transaction being committed.
- */
-export function isTransactionCommitChange(
-	change: SharedTreeBranchChange<unknown> & { type: "replace" },
-): boolean {
-	// "replace" is emitted by transaction commits and by rebase operations. A commit operation will
-	// always result in exactly one new commit. When a rebase results in exactly one new commit, we know
-	// that there will also be no removed commits:
+	// There is only one case in which a rebase both removes commits and adds exactly one new commit.
+	// This occurs when there is exactly one divergent, but equivalent, commit on each branch:
 	//
-	// A ─ B (branch X)
-	// └─ (branch Y)
+	// A ─ B (branch X)	  -- rebase Y onto X -->   A ─ B (branch X)
+	// └─ B' (branch Y)                                └─ (branch Y)
```

Comment:
I'm not sure why github renders this branching off of "A". It should appear to be branching off of B, as it is in the actual source code.